scripts/install_cli_tools.sh

```
@@ -41,44 +45,297 @@ extract_version() {
+    echo "error: neither sha256sum nor shasum is available; cannot verify lychee download" >&2
+    return 1
+  fi
+
+  local tmpdir
+  tmpdir="$(mktemp -d -t lychee-install.XXXXXX)"
+  # Use EXIT not RETURN: ``set -e`` aborts the script on a failed
+  # curl/sha256sum/tar before the RETURN trap would fire, leaking the
+  # temp dir. EXIT runs on both normal return AND set -e bailout.
+  # Double quotes expand ${tmpdir} now (function-local), since the
+  # trap body fires at script exit when the local has gone out of scope.
+  # shellcheck disable=SC2064 -- early expansion intentional: ${tmpdir} is function-local.
```

Shellcheck (0.11.0)

[error] Couldn't parse this shellcheck directive. Fix to allow more checks.

(SC1073)